tests/test_datamodel.py

```diff
@@ -0,0 +1,359 @@
+# This file is part of felis.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import os
+import unittest
+
+import yaml
+from pydantic import ValidationError
+
+from felis.datamodel import (
+    CheckConstraint,
+    Column,
+    DataType,
+    ForeignKeyConstraint,
+    Index,
+    Schema,
+    SchemaVersion,
+    Table,
+    UniqueConstraint,
+)
+
+TESTDIR = os.path.abspath(os.path.dirname(__file__))
+TEST_YAML = os.path.join(TESTDIR, "data", "test.yml")
+
+
+class DataModelTestCase(unittest.TestCase):
+    """Test loading of an entire test SDM schema from a YAML file."""
+
+    schema_obj: Schema
+
+    def setUp(self) -> None:
+        """Load data from test file."""
+        with open(TEST_YAML) as test_yaml:
+            data = yaml.safe_load(test_yaml)
+            self.schema_obj = Schema.model_validate(data)
```

Comment:
Do you really want the model to be validated in setup? Shouldn't the validation be in the test itself?

Reply:
I changed this to validate the model in a test method.